File: Completed/JavaScript/965.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isUnivalTree = function(root) {
  return helper(root, root.val)
};

function helper(root, val) {
  if (root === null) return true
  return root.val === val && helper(root.left, val) && helper(root.right, val)
}

File: Completed/Go/965.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func isUnivalTree(root *TreeNode) bool {
	return helper(root, root.Val)
}

func helper(root * TreeNode, v int) bool {
	if root == nil { return true }
	
	return root.Val == v && helper(root.Left, v) && helper(root.Right, v)
}